app/src/main/java/com/pvp/app/ui/screen/task/DatePicker.kt

```diff
@@ -0,0 +1,392 @@
+                }
+            )
+        }
+    }
+}
+
+@Composable
+fun TimePickerColumn(
+    range: IntRange,
+    selectedValue: Int,
+    onValueChange: (Int) -> Unit
+) {
+    val visibleItemCount = 3
+    val itemCount = range.count()
+    val extendedRange = (range.first - itemCount)..(range.last + itemCount)
+    val initialIndex = itemCount + selectedValue - (visibleItemCount / 2)
+
+    val listState = rememberLazyListState(initialFirstVisibleItemIndex = initialIndex)
+    LaunchedEffect(key1 = selectedValue) {
+        val scrollToIndex = itemCount + selectedValue - (visibleItemCount / 2)
+        listState.scrollToItem(scrollToIndex)
+    }
+
+    LaunchedEffect(key1 = listState) {
+        snapshotFlow { listState.firstVisibleItemIndex }
+            .map { firstVisibleItemIndex ->
+                val correctedIndex = (firstVisibleItemIndex % itemCount) + range.first
+                correctedIndex + (visibleItemCount / 2)
+            }
+            .distinctUntilChanged()
+            .collect { centeredValue ->
+                if (centeredValue in range) {
+                    onValueChange(centeredValue)
+                } else if (centeredValue == range.last + 1) {
+                    // After implementing infinite scroll, first value registers as last + 1.
+                    // Might be better way to address this, but works ig ¯\_(ツ)_/¯
+                    onValueChange(range.first)
+                }
+            }
+    }
+
+    Box(modifier = Modifier.height(50.dp * visibleItemCount)) {
+        LazyColumn(
+            state = listState,
+            modifier = Modifier.fillMaxWidth()
+        ) {
+            items(extendedRange.toList()) { value ->
+                val displayValue = when {
+                    value < range.first -> value + itemCount
+                    value > range.last -> value - itemCount
+                    else -> value
+                }
+                if (displayValue in range) {
+                    TimePickerItem(
+                        value = displayValue,
+                        isSelected = displayValue == selectedValue
+                    ) {}
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun TimePickerItem(value: Int, isSelected: Boolean, onItemSelected: () -> Unit) {
+    Box(
+        contentAlignment = Alignment.Center,
+        modifier = Modifier
+            .fillMaxWidth()
+            .height(50.dp)
+            .background(if (isSelected) MaterialTheme.colorScheme.primary else MaterialTheme.colorScheme.background)
+            .clickable { onItemSelected() }
+    ) {
+        Text(
+            text = "%02d".format(value),
+            style = MaterialTheme.typography.bodyLarge.copy(
+                color = if (isSelected) MaterialTheme.colorScheme.onPrimary else MaterialTheme.colorScheme.onBackground
+            )
+        )
+    }
+}
+
+@Composable
+fun ExpandableTimePicker(
+    transitionState: MutableTransitionState<Boolean>,
```

Comment:
Could be simply named `state`. 

Actually, all parameters that are not clarifying ones, could be just named for what they are used for. Here, in provided order, would be: `state`, `selected`, `onChanged`. If you agree with me, you could apply this for all the changes made in your PR to stay consistent.

In case there would be two parameters, i.e. `selected` that are used to provide selected date and time, then those should be clarified: `selectedDate` and `selectedTime`, **but not** `selected` and `selectedTime`.